File: frontend/src/types/tipoMembresia.ts
export interface TipoMembresia {
  id: number;
  nombre: string;
  duracionDias: number;
  precio: number;
  activa: boolean;
}

export interface CreateTipoMembresiaDto {
  nombre: string;
  duracionDias: number;
  precio: number;
}

File: GymApp/Models/TipoMembresia.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace GymApp.Models
{
    public class TipoMembresia
    {
        public int Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string Nombre { get; set; }

        [Range(1, 3650)]
        public int DuracionDias { get; set; }

        [Column(TypeName = "decimal(10,2)")]
        public decimal Precio { get; set; }

        public bool Activa { get; set; } = true;

        
        public List<Membresia> Membresias { get; set; } = new();
    }
}
